pkg/ingress/model_build_listener_rules.go

```diff
@@ -74,17 +77,46 @@ func (t *defaultModelBuildTask) buildListenerRules(ctx context.Context, lsARN co
 	return nil
 }
 
-// sort paths by length, if lengths equal, the path with Exact pathType has precedence
-func (t *defaultModelBuildTask) sortIngressPaths(paths []networking.HTTPIngressPath) []networking.HTTPIngressPath {
-	sort.Slice(paths, func(i, j int) bool {
-		if len(paths[i].Path) == len(paths[j].Path) {
-			if paths[i].PathType != nil {
-				return *paths[i].PathType == networking.PathTypeExact
+// sortIngressPaths will sort the paths following the strategy:
+// all exact match paths come first, no need to sort since exact match has to be unique
+// followed by prefix paths, sort by lengths - longer paths get precedence
+// followed by ImplementationSpecific paths or paths with no pathType specified, keep the original order
+func (t *defaultModelBuildTask) sortIngressPaths(paths []networking.HTTPIngressPath) ([]networking.HTTPIngressPath, error) {
+	exactPaths, prefixPaths, implementationSpecificPaths, err := t.classifyIngressPathsByType(paths)
+	if err != nil {
+		return nil, err
+	}
+	sortedPaths := exactPaths
+	sort.Slice(prefixPaths, func(i, j int) bool {
```

Comment:
sort.Slice is not stable and there is a potential for reversing the original order for equal length paths. Lets use sort.SliceStable instead

Reply:
Updated. Thanks